File: swift/tests/WalletKitTests/AtomicBlobStoreTests.swift
import Foundation
import XCTest
@testable import WalletKit

final class AtomicBlobStoreTests: XCTestCase {
    func testWriteReadDelete() throws {
        let root = makeTempDirectory()
        defer { try? FileManager.default.removeItem(at: root) }

        let store = IOSAtomicBlobStore(baseURL: root)
        let path = "account_keys.bin"
        let payload = Data([1, 2, 3, 4])

        try store.writeAtomic(path: path, bytes: payload)
        let readBack = try store.read(path: path)

        XCTAssertEqual(readBack, payload)

        try store.delete(path: path)
        let afterDelete = try store.read(path: path)
        XCTAssertNil(afterDelete)
    }
}

File: kotlin/walletkit-tests/src/test/kotlin/org/world/walletkit/AtomicBlobStoreTests.kt
package org.world.walletkit

import java.io.File
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNull

class AtomicBlobStoreTests {
    @Test
    fun writeReadDelete() {
        val root = tempDirectory()
        val store = FileBlobStore(root)
        val path = "account_keys.bin"
        val payload = byteArrayOf(1, 2, 3, 4)

        store.writeAtomic(path, payload)
        val readBack = store.read(path)
        assertEquals(payload.toList(), readBack?.toList())

        store.delete(path)
        assertNull(store.read(path))

        root.deleteRecursively()
    }
}
